Microsoft.Research/ManagedContract.Setup/buildMSI10.xml

```diff
@@ -370,8 +371,33 @@
 
     ==========================================================-->
   <Target Name="MergeExes"
-    DependsOnTargets="ILMergeFoxtrot;ILMergeAsmMeta;ILMergeClousot;ILMergeCCDoc"/>
+    DependsOnTargets="PrepareMerges;ExecuteMerges"/>
 
+  <Target Name="PrepareMerges" BeforeTargets="ExecuteMerges"
+    DependsOnTargets="RestoreToolPackages;PrepareILMerge"/>
+	
+  <Target Name="RestoreToolPackages">
+    <Exec Command="&quot;$(SolutionDir).NuGet\NuGet.exe&quot; restore &quot;$(SolutionDir)packages.config&quot; -PackagesDirectory &quot;$(SolutionDir)packages&quot;" />

```

Comment:
If you are going to build VSIX projects using \<MsBuild/\> task from within buildMSI10.xml or any other project file you do not need to run nuget from bat file. You can just use \<Exec/\> taks before \<MSBuild/\>. In that way any modification introduced by nuget packages will be processed by MsBuild after \<Exec/\>(nuget restore) finish.
Maybe we should talk about how the build process should be organized? Currently Editor Extension is not being built with Code Contracts. It would be best to have them build together. If we choose to use command line as a glue for them we should also consider Power Shell.